bundles/org.openhab.ui.basic/web-src/smarthome.js

```diff
@@ -1503,6 +1523,91 @@
 		_t.input.addEventListener("change", onChange);
 	}
 
+	/* class ControlInput extends Control */
+	function ControlInput(parentNode) {
+		Control.call(this, parentNode);
+
+		var
+			_t = this;
+
+		_t.input = _t.parentNode.querySelector("input[type=text]");
+		_t.itemType = _t.parentNode.getAttribute(o.itemTypeAttribute);
+		_t.verify = undefined;
+
+		var
+			lastValue = _t.input.value,
+			numberPattern = /^(\+|-)?[0-9\.,]+/,
+			dotSeparatorPattern = /^-?(([0-9]{1,3}(,[0-9]{3})*)|([0-9]*))?(\.[0-9]+)?$/,
+			commaSeparatorPattern = /^-?(([0-9]{1,3}(\.[0-9]{3})*)|([0-9]*))?(,[0-9]+)?$/;
+		function onChange() {
+			var
+				changeValue = _t.input.value,
+				changed = true;
+			if (_t.itemType === "Number") {
+				changeValue = changeValue.trim();
+				var numberValue = changeValue.match(numberPattern);
+				if (!numberValue || numberValue.length < 1) {
+					changed = false;
+				} else {
+					var unitValue = changeValue.substring(numberValue[0].length).trim();
+					changeValue = numberValue[0].replace(/^\+/, "");
+					if (commaSeparatorPattern.test(changeValue) && !dotSeparatorPattern.test(changeValue)) {
+						changeValue = changeValue.replace(/\./g, "").replace(",", ".");
+					}
+					if (unitValue.length > 1) {
+						changeValue = changeValue + " " + unitValue;
+					}
+				}
+			}
+
+			if (!changed) {
+				_t.setValuePrivate(lastValue);
```

Comment:
In case the item state was UNDEF and you enter an invalid number, this will lead to the field displaying an empty string rather than displaying `- °C` for example. Probably because the second parameter you do not provide is set to null.
To handle this case properly, this would require that you store the last value and + the last item state.

Reply:
In that case, the placeholder should show again as it online gets reset when something is received. The advantage of the placeholder is that it is a separate field, at the same place, only showing when the value is empty. The correction on empty string (removing the placeholder), is only relevant when receiving values.

Reply:
The problem is that you are resetting the input label when not expected, 
My solution avoids that. 
An even better solution would be to have a different method that only changes the input value.

Reply:
I will update my proposal later today.

Reply:
My new idea was not working.
In fact, I believe the code I proposed to you is just working perfectly.